dashboard/app/controllers/api/v1/pd/teacher_applications_controller.rb

```diff
@@ -30,12 +30,18 @@ def create
       application_json: application_json
     )
 
-    if @teacher_application.save
-      Pd::TeacherApplicationMailer.application_receipt(@teacher_application).deliver_now
-      Pd::TeacherApplicationMailer.principal_approval_request(@teacher_application).deliver_now
+    begin
+      if @teacher_application.save
+        Pd::TeacherApplicationMailer.application_receipt(@teacher_application).deliver_now
+        Pd::TeacherApplicationMailer.principal_approval_request(@teacher_application).deliver_now
+        head :no_content
+      else
+        render json: {errors: @teacher_application.errors.full_messages}, status: :bad_request
+      end
+    rescue ActiveRecord::RecordNotUnique
+      # Fail silently - this happens in the event of multiple submission
+      CDO.log.warn "Duplicate pd teacher application submission for user #{current_user.id} - ignoring the second one"
```

Comment:
It's really only the `save` line that will raise this error. I'd rather have begin/rescue blocks be as tight as possible. How about something like this?

```ruby
begin
  @teacher_application.save
rescue ActiveRecord::RecordNotUnique
  # Fail silently - this happens in the event of multiple submission
  CDO.log.warn "Duplicate pd teacher application submission for user #{current_user.id} - ignoring the second one"
  head :no_content
  return
end

if @teacher_application.persisted? 
  # send emails, etc.
else
  # render errors
end
```